rtos/TARGET_CORTEX/mbed_rtx_idle.cpp

```diff
@@ -108,8 +102,11 @@ static void default_idle_hook(void)
         // calculate how long we slept
         elapsed_ticks = os_timer->update_tick();
     }
-    svcRtxKernelResume(elapsed_ticks);
     core_util_critical_section_exit();
+
+    // Ensure interrupts get a chance to fire
+    __ISB();
```

Comment:
I think this ISB is overkill - we have SVC calls to osKernelResume and osKernelSuspend before our next disable, and SVC calls will allow interrupts to trigger according to the ARM ARM. (But then again, is it defined behaviour that these are SVC calls?)